File: standalone/src/main/java/io/delta/standalone/data/ActionCloseableIterator.java
package io.delta.standalone.data;

import io.delta.standalone.actions.Action;
import io.delta.standalone.internal.util.ConversionUtils;

import java.io.IOException;

public class ActionCloseableIterator implements CloseableIterator<Action> {

    CloseableIterator<String> stringIterator;

    public ActionCloseableIterator(CloseableIterator<String> stringIterator) {
        this.stringIterator = stringIterator;
    }

    @Override
    public boolean hasNext() {
        return stringIterator.hasNext();
    }

    @Override
    public Action next() {
        return ConversionUtils.convertAction(
                io.delta.standalone.internal.actions.Action.fromJson(stringIterator.next()));
    }

    @Override
    public void close() throws IOException {
        stringIterator.close();
    }
}

File: standalone/src/main/scala/io/delta/standalone/internal/data/ActionCloseableIterator.scala
package io.delta.standalone.internal.data

import io.delta.standalone.actions.Action
import io.delta.standalone.data.CloseableIterator

import io.delta.standalone.internal.util.ConversionUtils

case class ActionCloseableIterator(stringIterator: CloseableIterator[String])
  extends CloseableIterator[Action]{

  override def hasNext: Boolean = {
    stringIterator.hasNext
  }

  override def next(): Action = {
    ConversionUtils.convertAction(
      io.delta.standalone.internal.actions.Action.fromJson(stringIterator.next))
  }

  override def close(): Unit = {
    stringIterator.close()
  }
}
